### PR TITLE
Create CI pipeline for ingress-node-firewall repo

### PR DESCRIPTION
**Pull request by msherif1234**
Signed-off-by: Mohamed Mahmoud <mmahmoud@redhat.com>

### CODE REVIEW

File: ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  name: "4.12"
+  namespace: ocp
+tests:
+- as: e2e-aws
+  steps:
+    cluster_profile: aws
+    workflow: openshift-e2e-aws
+- as: ingress-node-firewall-e2e-metal-ipi
+  steps:
+    cluster_profile: packet
```
Comment (martinkennelly) [inline ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: do you know what this indicates?

**Comment by martinkennelly** [comment]
/lgtm
Not an expert with the CI config files but didn't spot anything unusual.

File: ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml (hunk 1 above)
Comment (andreaskaris) [reply ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: That triggered this rehearsal here:
* https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/30687/rehearse-30687-pull-ci-openshift-ingress-node-firewall-master-ingress-node-firewall-e2e-metal-ipi/1550213014077050880
So not a metal-ipi job

Comment (andreaskaris) [reply ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: So it just builds the packet server
~~~
[akaris@linux release (aws-serial-for-ovn-kubernetes)]$ cat ./ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-ref.yaml 
(...)
  documentation: |-
    This step performs the setup of the packet server
~~~

Comment (msherif1234) [reply ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: test e2e with traffic on BM again this is just place holder will have more content once we get to E2E implementation

Comment (andreaskaris) [reply ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: TL;DR: if we just want to run *any* tests, then /lgtm.

But the baremetal test right now tests nothing and the aws test right now just runs the e2e suite.

-----------------------------

Would a real baremetal test be something like:
~~~
  71 - as: e2e-metal-ipi-ovn-dualstack
   72   optional: true
   73   steps:
   74     cluster_profile: equinix-ocp-metal
   75     env:
   76       DEVSCRIPTS_CONFIG: |
   77         IP_STACK=v4v6
   78         NETWORK_TYPE=OVNKubernetes
   79     workflow: baremetalds-e2e
~~~

I'm pretty sure we have to add tests to the serial test suit anyway in origin/, or define an entirely new test suite?

So for aws:
~~~
- as: e2e-aws-serial
  steps:
    cluster_profile: aws-2
    env:
      BASE_DOMAIN: aws-2.ci.openshift.org
      TEST_SUITE: openshift/conformance/serial
    workflow: openshift-e2e-aws-serial
~~~

for baremetal:
~~~
- as: e2e-metal-ipi-serial
  optional: true
  steps:
    cluster_profile: equinix-ocp-metal
    workflow: baremetalds-e2e-serial-ipv4
~~~

Comment (msherif1234) [reply ci-operator/config/openshift/ingress-node-firewall/openshift-ingress-node-firewall-master.yaml:51]: correct this is just place holder till we have E2E as I said above we need to enable ci bot for the new repo and this is more or less template that we will fill in along the way

**Review by droslean — APPROVED** [review]

**Pull request merged**